server/protocol.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	curppb "github.com/xline-kv/mock-xline/gen/curppb"
+	xlinepb "github.com/xline-kv/mock-xline/gen/xlinepb"
+	"github.com/xline-kv/mock-xline/sample"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+type protocolServer struct {
+	curppb.ProtocolServer
+
+	fetchClusterSample sample.FetchClusterMap
+	proposeSample      sample.ProposeMap
+	waitSyncSample     sample.WaitSyncedMap
+	commandSample      sample.CommandMap
+}
+
+func NewProtocolServer(
+	fetchClusterSample sample.FetchClusterMap,
+	proposeSample sample.ProposeMap,
+	waitSyncSample sample.WaitSyncedMap,
+) *protocolServer {
+	return &protocolServer{
+		fetchClusterSample: fetchClusterSample,
+		proposeSample:      proposeSample,
+		waitSyncSample:     waitSyncSample,
+		commandSample:      sample.CommandMap{},
+	}
+}
+
+func (s *protocolServer) FetchCluster(ctx context.Context, req *curppb.FetchClusterRequest) (*curppb.FetchClusterResponse, error) {
+	fmt.Println("fetch cluster", req)
```

Comment:
Could using logging be better than println?

Reply:
We plan to assert the output in a later update. So use fmt for output temporary